opentelemetry-otlp/examples/basic-otlp/src/main.rs

```diff
@@ -59,33 +60,37 @@ fn init_logs() -> SdkLoggerProvider {
         .build()
 }
 
-#[tokio::main]
-async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
-    let logger_provider = init_logs();
+thread_local! {
+    static SUPPRESS_GUARD: RefCell<Option<opentelemetry::ContextGuard>> = const { RefCell::new(None) };
+}
+
+// #[tokio::main]
```

Comment:
todo: cleanup http one and confirm it does not need this technique by default.
todo: see if the client authors offer a way to opt-out of telemetry.